### PR TITLE
Feat / Add periodic task refetching to campaign page

### PR DESCRIPTION
**Pull request by royschut**
This PR:
- adds periodic refetching once there are no tasks (anymore)
- adds filtering to make sure submitted tasks aren't refetched
- refactoring: uses useLazyQuery for fetching instead of client.query for better readability

### CODE REVIEW

**Review by RCVZ — APPROVED** [review]
Nice!

File: src/containers/Task/Task.jsx [hunk 1]
```diff
@@ -2,139 +2,102 @@ import React, { useState, useEffect } from 'react';
...
-    setTaskIdentifiers(remainingTaskIdentifiers);
-    if (remainingTaskIdentifiers.length > 0) {
-      history.replace(`/campaign/${campaignIdentifier}/task/${remainingTaskIdentifiers[0]}`);
+  const history                             = useHistory();
+  const [campaign, setCampaign]             = useState(null);
+  const [task, setTask]                     = useState(null);
+  const [tasksTodo, setTasksTodo]           = useState([]);
+  const [tasksSubmitted, setTasksSubmitted] = useState([]);
+  const taskDrawAmount                      = 10;
+  const refetchTimeout                      = 3000;
```
Comment (VincentCouzij) [inline src/containers/Task/Task.jsx]: Is this not a bit much? do we expect new tasks to be added every 3 seconds?

Comment (royschut) [reply src/containers/Task/Task.jsx]: That was for testing purposes, but thanks for reminding me. 30000 it is!

File: src/containers/Task/Task.jsx [hunk 2]
```diff
@@ -2,139 +2,95 @@ import React, { useState, useEffect } from 'react';
...
-    if (remainingTaskIdentifiers.length > 0) {
-      history.replace(`/campaign/${campaignIdentifier}/task/${remainingTaskIdentifiers[0]}`);
+  const history                             = useHistory();
+  const [campaign, setCampaign]             = useState(null);
+  const [task, setTask]                     = useState(null);
+  const [tasksTodo, setTasksTodo]           = useState([]);
+  const [tasksSubmitted, setTasksSubmitted] = useState([]);
+  const taskDrawAmount                      = 10;
+  const refetchTimeout                      = 30000;
+  const [refetchCount, setRefetchCount]     = useState(0);
```
Comment (demmyhonore) [inline src/containers/Task/Task.jsx:19]: Are refetchCount and taskDrawAmount doing two different things?

File: src/containers/Task/Task.jsx [hunk 3]
```diff
@@ -2,139 +2,95 @@ import React, { useState, useEffect } from 'react';
...
-        });
-    }
-  }, [campaignIdentifier, history, taskIdentifier, taskTotal, tasksReceived]);
-
-  // Retrieve tasks on new draw (task identifier not in url)
-  useEffect(() => {
-    if (taskIdentifier || loading || !campaign) {
-      return;
+    if (!taskIdentifier && score) {
+      getTaskList({ variables: { identifier: score.identifier, filtered: tasksSubmitted, drawamount: taskDrawAmount } });
```
Comment (demmyhonore) [inline src/containers/Task/Task.jsx]: Is it required to make drawamount camelCasing here?

**Review by demmyhonore — APPROVED** [review]
Nice refactor, a few minor questions.

File: src/containers/Task/Task.jsx (hunk 2 above)
Comment (royschut) [reply src/containers/Task/Task.jsx:19]: Yes. The taskDrawAmount is the maximum amount of tasks fetched. The refetchCount is a counter that adds up on every refetch timeout (30secs), so the useEffect with the fetch gets retriggered.

File: src/containers/Task/Task.jsx (hunk 3 above)
Comment (royschut) [reply src/containers/Task/Task.jsx]: Good point. It's not required, but it's allowed, so it makes sense to keep consistency

**Review by ChristiaanScheermeijer — APPROVED** [review]

**Pull request merged**